crates/sui-types/src/gas.rs

```diff
@@ -383,8 +383,14 @@ impl<'a> SuiGasStatus<'a> {
     pub fn bucketize_computation(&mut self) -> Result<(), ExecutionError> {
         let computation_cost: u64 = self.gas_used().into();
         let bucket_cost = get_bucket_cost(&COMPUTATION_BUCKETS, computation_cost);
-        let charge = bucket_cost - computation_cost;
-        self.deduct_computation_cost(&charge.into())
+        let charge = bucket_cost.saturating_sub(computation_cost);
+        if charge > 0 {
+            self.deduct_computation_cost(&GasUnits::new(charge).to_unit())
+        } else {
+            // we hit the last bucket and the computation is already more then the
+            // max bucket so just charge as much as it is without buckets
```

Comment:
II find this comment a little confusing--it sounds like we might charge more than MAX_BUCKET_COST, but if I read the code correctly we'll hit this branch exactly when `computation_cost == MAX_BUCKET_COST`. Might be clearer to add that as a `debug_assert` if that's the case.

Reply:
well this may be a bit strange or related to other conversations.
If you look at the last bucket that is
`ComputationBucket::new(50_001, u64::MAX, MAX_BUCKET_COST),`
where `const MAX_BUCKET_COST: u64 = 1_000_000;` so the idea here is that if for whatever reason you manage to get in that bucket you are charged `1_000_000` which is fine (meaning it's a number we defined for now).
However the bucket is covering for all u64s. So it may happen that the value is bigger than `MAX_BUCKET_COST` in which case we do not want to charge less but we want to leave things as they are.
That is probably a case that can be covered by making the max amount of gas being the value in the last bucket (which may even be the case now) but it seems safer to me to have the code as is now given that it should not cause any problem and do the right thing.
Thoughts?

Reply:
Works for me.